src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1385,5 +1402,49 @@ protected override Cci.IMethodDefinition CreatePrivateImplementationDetailsStati
         {
             return new SynthesizedPrivateImplementationDetailsStaticConstructor(SourceModule, details, GetUntranslatedSpecialType(SpecialType.System_Void, syntaxOpt, diagnostics));
         }
+
+        internal virtual SynthesizedAttributeData SynthesizeEmbeddedAttribute()
+        {
+            // Embedded attributes should never be synthesized in modules.
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        internal SynthesizedAttributeData SynthesizeIsReadOnlyAttribute(Symbol symbol)
+        {
+            if ((object)Compilation.SourceModule != symbol.ContainingModule)
+            {
+                // For symbols that are not defined in the same compilation (like NoPia), don't synthesize this attribute.
+                return null;
+            }
+
+            return TrySynthesizeIsReadOnlyAttribute(symbol);
+        }
+
+        protected virtual SynthesizedAttributeData TrySynthesizeIsReadOnlyAttribute(Symbol symbol)
+        {
+            if ((object)Compilation.SourceModule != symbol.ContainingModule)
```

Comment:
Yes. A refactoring issue that I should've caught earlier :(
Thanks for pointing it out.

---
In reply to: [115858076](https://github.com/dotnet/roslyn/pull/18715#discussion_r115858076) [](ancestors = 115858076)